### DISCUSSION TITLE
Using Auto3DSeg output with MONAI Label

### DISCUSSION THREAD

**Discussion by landynfrancis** [Project-MONAI/tutorials#1714] [Q&A] [answered]
Hello,

I have started to use Auto3DSeg, and I am very impressed with it's performance. I understand it outputs MONAI Bundles. I was under the impression that MONAI Label could be configured to use MONAI Bundles, but the structure of the Auto3DSeg bundles looks different from other MONAI Label bundles I have used.

Is there a way I can take an algorithm created using Auto3DSeg and use it with MONAI Label to execute active learning?

At the very least it would be nice if I could export the network setup so I can use the .pt file to load weights. 

Any advice and guidance would be appreciated.

Thank you

**Comment by diazandr3s** [comment, marked as answer]
Hi @landynfrancis,

Very good question.

Currently, Auto3DSeg models can't be consumed in MONAI Label. Some changes have to be made in the radiology app (segmentation model) before using the pre-trained model from Auto3DSeg. Specifically the `lib/configs/segmentation.py` and `lib/infers/segmentation.py` files
As an example, I've modified the MONAI Label radiology app (segmentation model) for the ICrH model available in the [Slicer MONAI Auto3DSeg](https://github.com/lassoan/SlicerMONAIAuto3DSeg) module - This ICrH model was trained using the SegResNet architecture.

Here you can see the radiology folder and a video showing how it works.

[radiology.zip](https://github.com/Project-MONAI/tutorials/files/15345214/radiology.zip)

https://github.com/Project-MONAI/tutorials/assets/11991079/669791f9-0b40-41d2-98af-5c9ee8fe0ed5

The `model.pt` file from `segresnet_0/model` should be renamed to **_pretrained_segmentation-anatomy_and_icrh.pt_** and placed in the `radiology/model` folder

Currently, this is only for inference.

Hope this helps as inspiration.

**Reply by klchtsk** [reply]
Dear @diazandr3s ! I have also used your code to use MONAI Auto3DSeg model for MONAI Label. As you mention, it's not working for model training. What changes should be made for training to be executed correctly? DataLoader seems to be outputting list of individual items not batch of tensors. Thank you!

**Reply by diazandr3s** [reply]
Hi @klchtsk,

Thanks for commenting here.

Unfortunately, Auto3DSeg training feature isn't implemented in MONAI Label - only inference. For training or retraining the Auto3DSeg you should follow the Auto3DSeg instructions. 

Hope this helps,

**_EDIT:_** If you mean using the SegResNet architecture instead of the whole Auto3DSeg system, that's already working in MONAI Label: https://github.com/Project-MONAI/MONAILabel/blob/e31eaee03b4a3e480532de0e01a1d96feef799e1/sample-apps/radiology/lib/configs/segmentation.py#L82-L90
